### ISSUE TITLE
Error failed to load extension on Ruby 3.0.0 that doesn't happen on Ruby 2.7.2

### ISSUE THREAD

**Issue by johannesluedke** [appsignal/appsignal-ruby#673]
## Describe the bug

Running the tests of my short example app with on Ruby 2.7.2 works, but on Ruby 3.0.0 yields `ERROR: AppSignal failed to load extension. Please run `appsignal diagnose` and email us at support@appsignal.com
LoadError: cannot load such file -- appsignal_extension`

## To Reproduce

- Using AppSignal for Ruby gem version  2.11.1
- With this code:
```
require 'appsignal'

class MyJob
  def work(msg)
    Appsignal.monitor_transaction(
      'perform_job.my_job',
      class: self.class.name,
      method: __method__.to_s,
    ) { puts "working }
  end
end
  ```
(In the real world app where I saw the issue first this would be a Sneaker worker, but the same problem occurs with this minimal example)

**Comment by thijsc** [comment]
We have a pull request with some minor changes for 3.0 going on in #671.

**Comment by johannesluedke** [comment]
PR is now https://github.com/appsignal/appsignal-ruby/pull/681

**Comment by tombruijn** [comment]
Fix released as part of Ruby gem 2.11.2. Let us know if you run into any issues. Closing for now.
